crates/ln-dlc-node/src/node/wallet.rs

```diff
@@ -104,4 +108,57 @@ impl Node {
             pending_close,
         }
     }
+
+    pub fn get_on_chain_history(&self) -> Result<Vec<bdk::TransactionDetails>> {
+        self.wallet
+            .on_chain_transactions()
+            .context("Failed to retrieve on-chain transaction history")
+    }
+
+    pub fn get_off_chain_history(&self) -> Vec<PaymentDetails> {
+        let inbound_payments = self
+            .inbound_payments
+            .lock()
+            .expect("to be able to acquire lock");
+        let inbound_payments = inbound_payments.iter().map(|(hash, info)| PaymentDetails {
+            payment_hash: *hash,
+            status: info.status,
+            flow: PaymentFlow::Inbound,
+            amount_msat: info.amt_msat.0,
+            timestamp: info.timestamp,
+        });
+
+        let outbound_payments = self
+            .outbound_payments
+            .lock()
+            .expect("to be able to acquire lock");
+        let outbound_payments = outbound_payments.iter().map(|(hash, info)| PaymentDetails {
+            payment_hash: *hash,
+            status: info.status,
+            flow: PaymentFlow::Outbound,
+            amount_msat: info.amt_msat.0,
+            timestamp: info.timestamp,
+        });
+
+        let mut payments = inbound_payments
+            .chain(outbound_payments)
+            .collect::<Vec<_>>();
+
+        payments.sort_by(|a, b| b.timestamp.cmp(&a.timestamp));
```

Comment:
Same as for on-chain transactions: This sorting is a bit superfluous for the task at hand because we later combine all the transactions, payments and trades and sort them again. But I thought it might be good to sort this regardless in case we ever need to consume the list of off-chain payments separately.